location/models.py

```diff
@@ -160,12 +160,33 @@ def get_location_from_ids(self, qsr, loc_type):
         return list(qsr)
     
     def is_allowed(self, user, locations_id):
+        if user.is_superuser or not settings.ROW_SECURITY:
+            return True 
         if hasattr(user, '_u'):
             user = user._u
         cache_name = f'user_locations_{user.id}'
         allowed = cache.get(cache_name)
         if not allowed:
-            allowed = list(self.allowed(user.id, qs=None))
+            # for CA
+            if user.is_claim_admin and user.health_facility:
+                allowed = list(self.children(
+                    user.id, 
+                    location_id=user.health_facility.location_id
+                ))
+            if user.is_officer:
+                villages = list(OfficerVillage.objects.filter(
+                    officer=Officer.objects.filter(
+                        code=user.username,
+                        *filter_validity()
+                    ),
+                    *filter_validity()
+                ).values_list(location_id, flat=True))
+                allowed = list(self.parent(
```

Comment:
@delcroip LocationManager only has `parents` defined but here it's`self.parent` without s, typo or intentional? I see `Officer.officer_allowed_locations` implementation might be problematic as the intention seems to be looking up two levels of parents and add them both to allowed locations but only the first level of parent got added.

Reply:
Might be worth adding tests for Officer permission to clarify

Reply:
sure, I  planned that for tomorrow, on Friday I wanted to make sure the individual part worked